### DISCUSSION TITLE
Upgrade the README on atom.io/packages

### DISCUSSION THREAD

**Discussion by mauricioszabo** [atom/atom#22904] [Q&A]
So, recently (I'm not really _that sure_ that's recently, but anyway) I've been unable to reliably update my README on atom.io/package. Yes, it sometimes work, but sometimes don't, the images get all wrong links and I'm not sure why, and the whole thing gets worse when you have a package that is not on GitHub.

For example, here's one example: https://atom.io/packages/simple-git
Here's another: https://atom.io/packages/chlorine

No images appear at all, and they do appear both on Github and on Gitlab's pages. So, how to make it work?

**Comment by the-j0k3r** [comment]
I would maybe use a cdn for links to images posted on gitlab .

e.g. https://raw.githack.com/

I have same issue on our projects https://atom.io/packages/terminus https://atom.io/packages/x-terminal, so something is preventing the images from rendering.

**Comment by nlydv** [comment]
Apparently this image issue has been going on for around a year now... which is pretty ridiculous if you ask me.
I found a simple workaround with just a few lines of JS about 6 months back and made a basic browser extension out if it.

Chrome Webstore: [Atom.io — Image Fixer](https://chrome.google.com/webstore/detail/atomio-%E2%80%94-image-fixer/dckhejcdhaojgaafhjmeengkkilgejma)
Source Repo: [nlydv/atom-io-imgfix](https://github.com/nlydv/atom-io-imgfix)

Enjoy!

**Reply by the-j0k3r** [reply]
it be better to fix it upstream after the fact, since its not available for all browsers.

Someone is probably working on the vscode store instead

**Reply by nlydv** [reply]
Well yeah obviously it'd be better if they just fixed the issue at the source... the extension is very simple though and should be compatible with Firefox and other browsers that use similar web extensions APIs. Just haven't formally submitted it to Firefox's extension store.

**Reply by the-j0k3r** [reply]
I get it, but whats the point I ask, I'm interested as developer (because I have published atom packages) not as a user, and most users wont come here or look at your extension.

So I'de like my users to see what those images are.

Now extension is edge cases, I cant tell people who use our packages to go and download your extension, I rather just redirect them to the real readme.

Im sure their atom io site has a source code repo somewhere and its opensource, so a contrib is most likely to be the best way, i.e. fix it for all not just a hand full.

Sorry not for me. This should be a contribution to their source code as a fix.

**Reply by nlydv** [reply]
I've looked and it doesn't seem like their site is open source 😕

**Comment by the-j0k3r** [comment]
Ive reported this as an issue upstream https://github.com/atom/atom.io/issues/225

Turns out that the site has no public source code repo, which makes for an interesting way to fix any issues as a user.

Let me ping @darangi hes usually very helpful and maybe can redirect the issue to the devs and finally get this fixed.

**Reply by darangi** [reply]
@the-j0k3r  @mauricioszabo  we will take a look into the situation

**Reply by afmarchetti** [reply]
any news in this? is riducolus you want that users switch to vscode?

**Reply by the-j0k3r** [reply]
People sure have bigger priorities, not only in real live but professionally, also likely someone with access to the relevant system is required.

In the grand scheme of things images on readme's aren't a crucial portions to the normal functioning of the system, which still works and people can always visit the projects repo.

Its annoying but not worth the bother.

**Reply by mauricioszabo** [reply]
@the-j0k3r The trouble is, someone already made the code for them. There's a Chrome extension that solves this issue. There are two issues to solve this: the one you opened, and the one that I opened here: https://github.com/atom/atom/issues/22929

I pointed on my issue how to solve the problem, down to the very JS that solves the issue - it's 9 JS lines of code, but on their repository could be even less, because the issue is just to swap the properties `data-canonical-src` to `src`. On September, @sadick254 commented that "it was on their radar, and it'll be solved as soon as they can" - it sure is a looong time to swap 2 attributes, right?

I can only see two options: one, they lost the source code or the deploy instructions for the `atom.io/packages` somehow, or they simply don't care and actually want to see Atom being discontinued. I honestly bet on the second choice

**Reply by the-j0k3r** [reply]
@mauricioszabo 

None of the people you mention have access to the server side to change anything, none work for GitHub/Microsoft.

Its the same as asking you to go change the 2 lines yourself.

As for the extension, asking someone to install some extension, its the same as asking people to click on the repo link on the atom.io to go read the readme github repo page, the later its faster actually and no extensions necessary and works on all browsers, as not everyone uses Chrome. FYI I wouldn't install an extension that for the sake of what amounts to saving ppl one click to open a new tab and see pretty images.

Im not defending anyone mind, these are the facts.

Microsoft you know as well as I do, its wheels within wheels, too many Chiefs and not enough Indians.

Send them a letter snail mail type complain, Open a petition online and ask people to sign it (you need 100K signatures for it to be heard)

Now, Ill go back to my real life problems, I have much bigger problems that cant be solved by posting on GitHub.

**Comment by afmarchetti** [comment]
@the-j0k3r ok go to your real life i will switch to vscode today, atom will be dead in 1 year (it's sad)

**Reply by the-j0k3r** [reply]
@afmarchetti Do whatever you like, freedom is as one can do, I'll continue using Atom and then when it no longer works, I will use the community edition if it comes to be https://github.com/atom-community/atom

If it doesn't come to be, I will find another editor other than VScode (its closed source nonsense), which by then I wont even be running Windows at all and no more MS products for me. Plenty choices out there, even replacements for GitHub.

Good luck.
